File: Java/896. Monotonic Array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        if (nums.length < 3)    return true;
        
        int up = 0, down = 0;
        for (int i=1; i<nums.length; i++) {
            if (nums[i] > nums[i-1])    up = 1;
            else if (nums[i] < nums[i-1])   down = 1;
            
            if (up + down > 1)  return false;
        }
        
        return true;
    }
}

File: Python3/896. Monotonic Array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        # edge cases
        if len(nums) < 3:
            return True
        
        # initial state
        up = down = 0
        
        for i, n in enumerate(nums[1:], 1):
            # nothing to do 
            if n == nums[i-1]:
                continue
            
            # increasing
            elif n > nums[i-1]:
                up = 1
            
            # decreasing
            else:
                down = 1
            
            if up + down > 1:
                return False
            
        return True